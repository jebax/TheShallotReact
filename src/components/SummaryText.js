import React from 'react'
import PropTypes from 'prop-types'

const SummaryText = ({ sentences }) => {
  const summaryBody= (() => {
    if (sentences.length === 0) {
      return (
        <span
          className='loadingText'
        >
          Loading summary...
        </span>
      )
    } else {
      return sentences.map((sentence, index) => {
        return (
          <p
            className='summarySentence'
            key={index}
          >
            {sentence}
          </p>
        )
      })
    }
  })()

  return (
    <article
      className='summaryText'
    >
      {summaryBody}
    </article>
  )
}

SummaryText.propTypes = {
  sentences: PropTypes.array.isRequired
}

export default SummaryText
