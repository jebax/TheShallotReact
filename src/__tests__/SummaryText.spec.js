import React from 'react'
import { shallow } from 'enzyme'
import SummaryText from '../components/SummaryText'

describe('<SummaryText />', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(
      <SummaryText
        sentences={['One', 'Two', 'Three', 'Four', 'Five']}
      />
    )
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders each sentence', () => {
    const summaryText = wrapper.find('.summary-text')

    expect(summaryText.text()).toEqual('OneTwoThreeFourFive')
  })

  it('displays a "loading" message if there are no sentences loaded', () => {
    const secondWrapper = shallow(<SummaryText sentences={[]}/>)
    const summaryText = secondWrapper.find('.summary-text')

    expect(summaryText.text()).toEqual('Loading summary...')
  })
})
