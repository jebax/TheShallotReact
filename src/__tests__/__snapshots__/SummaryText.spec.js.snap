// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryText /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SummaryText
    sentences={
      Array [
        "One",
        "Two",
        "Three",
        "Four",
        "Five",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="summarySentence"
        >
          One
        </p>,
        <p
          className="summarySentence"
        >
          Two
        </p>,
        <p
          className="summarySentence"
        >
          Three
        </p>,
        <p
          className="summarySentence"
        >
          Four
        </p>,
        <p
          className="summarySentence"
        >
          Five
        </p>,
      ],
      "className": "summaryText",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "One",
          "className": "summarySentence",
        },
        "ref": null,
        "rendered": "One",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Two",
          "className": "summarySentence",
        },
        "ref": null,
        "rendered": "Two",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Three",
          "className": "summarySentence",
        },
        "ref": null,
        "rendered": "Three",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "Four",
          "className": "summarySentence",
        },
        "ref": null,
        "rendered": "Four",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "Five",
          "className": "summarySentence",
        },
        "ref": null,
        "rendered": "Five",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="summarySentence"
          >
            One
          </p>,
          <p
            className="summarySentence"
          >
            Two
          </p>,
          <p
            className="summarySentence"
          >
            Three
          </p>,
          <p
            className="summarySentence"
          >
            Four
          </p>,
          <p
            className="summarySentence"
          >
            Five
          </p>,
        ],
        "className": "summaryText",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "One",
            "className": "summarySentence",
          },
          "ref": null,
          "rendered": "One",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Two",
            "className": "summarySentence",
          },
          "ref": null,
          "rendered": "Two",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Three",
            "className": "summarySentence",
          },
          "ref": null,
          "rendered": "Three",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "Four",
            "className": "summarySentence",
          },
          "ref": null,
          "rendered": "Four",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "Five",
            "className": "summarySentence",
          },
          "ref": null,
          "rendered": "Five",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
